### ISSUE TITLE
Some Meta Stuff

### ISSUE THREAD

**Issue by benjamin-kirkbride** [Akuli/porcupine#1324]
Can we enable Github Discussions? I would have put this "issue" in there, as well as some other thoughts that I've had. Trying not to pollute the issue pile too much (believe it or not) :)
---
Can we get some new tags:
- `Performance` for issues relating to performance
- `Meta` for issues (like this one) that mostly relate to how the project is organized, ran, discussed, etc
---
What kind of efforts have been made to promote Porcupine? I think we should consider posting to `/r/python` on each major release. Maybe to HN as well.

I also think that the value positioning of this project should be tweaked slightly in the README. The list of features is not a bad thing to have, but obviously going head to head on features against basically any other Editor/IDE that anyone would have heard of, Porcupine does not hold a candle.

IMO the thing that is exciting about Porcupine is that it is very easy to extend using Python I went from 0 to having a Markdown Plugin written in like 8 hours? I don't think there is any other Python IDE out there that comes remotely close. Even PyCharm uses not-Python for it's extensions (Java). Beyond that, everything about Porcupine is extensible and changeable, fairly easilly. VSCode obviously has way more extensions, but there are a ton of things you CAN'T change about it, because they don't expose the API you would need. This makes VSCode a poor choice for a lot of applications.

I think we should hit the points above early, and hard in the docs. Make it clear that Porcupine is THE editor for people who want to hack on their editor in Python, first and foremost. Then discuss how it is a pretty good editor broadly.

**Comment by Akuli** [comment]
> Github discussions

There are already two places to discuss Porcupine things, IRC for casual things and github issues for anything that you want all Porcupine developers to see. I think this is nice and simple, and I don't really see a need for a third place. Also "pollute the issue pile" is not much of a problem IMO, since it is a big pile anyways. But I would like to hear what @rdbende and @Moosems think.

> new tags

I added a performance tag. I don't think we need a meta tag since there aren't very many meta issues, and they tend to get closed quickly because there's no programming/debugging work involved.

> to promote Porcupine?

I haven't done much. For a few years I have thought that I will promote Porcupine more once I get the most annoying issues fixed, and I still think so. Currently "the most annoying issues" means at least #560 and #921.

> README

I agree with everything you said about README. Feel free to make a PR :)

**Comment by Moosems** [comment]
I brought this up once in the past and while IRC is nice it's hard to use on the go and to leave messages. GitHub discussions are also often left alone and I've waited for a while for answers by using issues before. In fact I still have some questions that haven't been answered on a private repo. As I said in the past I truly believe that using a Discord   Channel would help out a lot.

**Comment by Moosems** [comment]
Plus, a Discord server helps create a strong community around Porcupine and allows for reading others messages in the past which means no knowledge is ever lost to the logs that eventually get overwritten on some systems.

**Comment by Akuli** [comment]
IRC is more in Porcupine's "very easy to extend" style than discord, as it is very easy to write bots and custom clients and the like for it. But I also see the advantages of discord. Maybe the move to discord should happen at about the same time we decide to promote Porcupine more.

And speaking of IRC, I won't have time for IRC today. Maybe tomorrow :)

**Comment by rdbende** [comment]
> Can we get some new tags:
>
>    Performance for issues relating to performance
>    Meta for issues (like this one) that mostly relate to how the project is organized, ran, discussed, etc

That's fine for me, and I think it's fine for Akuli, too, there's really a need for a "performance" label next to "shitty-highlighter" these days :))


> What kind of efforts have been made to promote Porcupine? I think we should consider posting to /r/python on each major release. Maybe to HN as well.

AFAIK there hasn't been such effort. Personally I found @Akuli and this project after looking at the contributors of TkinterEP, however I can imagine Akuli telling/forcing his friends to use Porcupine XDD

That said I think that we should promote it in some way, because Porcupine is an awesome editor, but not many people are aware of it.

**Comment by rdbende** [comment]
Lol, you were faster than me :D

**Comment by rdbende** [comment]
> In fact I still have some questions that haven't been answered on a private repo.

Uhh, that's for me :), sorry.
You're @mentioning me so much that I can't even sort out the more important ones from my notifications, so I'd rather get on with my own stuff in that little free time I have.

You won't see me for the next 10 days, but I'll try to check out your questions now.

**Comment by Akuli** [comment]
In general, I agree that sometimes discussions remain unanswered forever, but the same can be said about github issues or discord servers or anything else. It's just how open-source works, especially in projects with lots of confused users and only a few developers.

**Comment by benjamin-kirkbride** [comment]
The benefit of Github Discussions (and Discord Forums) vs GH Issues and IRC is they are not ephemeral. They don't "close" or get lost in the same way. You can have a "What do you like to use Porcupine for?" thread that lasts forever.

It also makes the issues less of a dumping ground for everything including support type questions. At the current scale, this is obviously not an issue, but imagine if there were 10x the number of people using it.

**Comment by benjamin-kirkbride** [comment]
> I agree with everything you said about README. Feel free to make a PR :)

I'll add it to the list ;)

**Comment by Moosems** [comment]
> Uhh, that's for me :), sorry.
> 
> You're @mentioning me so much that I can't even sort out the more important ones from my notifications, so I'd rather get on with my own stuff in that little free time I have.


I @'d you @rdbende? Blasphemous 🤣!

**Comment by Moosems** [comment]
Discord allows easy creation of bots in Python and now has a large enough user base that it'd be worth it. It also simplified image and video sharing along with being more accessible for more people these days.

**Comment by benjamin-kirkbride** [comment]
> promote Porcupine more once I get the most annoying issues fixed

Can we get a tag for this? `embarrassing` or `blocks-promotion` or something?

**Comment by Akuli** [comment]
I added a new `embarrassing` tag. I like the name :)

**Comment by Moosems** [comment]
Me and ThePhilgrim are both on Discord. @benjamin-kirkbride I could add you as a friend and start a channel. @Akuli and @rdbende Using Discord has the advantage of being able to discern among the many emails you may get from Github and allows for someone to more easily contact you for something possibly time sensitive.

**Comment by benjamin-kirkbride** [comment]
I am `beefy_swain` on Discord.

**Comment by Moosems** [comment]
Friend request sent.

**Comment by Moosems** [comment]
We can still use IRC but having a Discord server set up would be super useful for contacting and seeing past messages from when you were gone.

**Comment by rdbende** [comment]
Honestly, I still don't get the hype around Discord, but if you want, and if I can find a decent (not Electron based) desktop client, I can create an account.
However I'm busy with work, so I wouldn't really be available on Discord either.

The good thing with Github email notifications, that I can download my mails, and then compose answers on the train, where I don't have internet connection anyways.

**Comment by Moosems** [comment]
It has builtin file, image, and video sharing and allows for looking at past messages from others.I don't care about hype for it but find it quite practical on many occasions. Also, you have stated before finding trouble in finding important notifications due to my mass-commenting problem and the many GH actions.

**Comment by Moosems** [comment]
Plus we have ThePhilgrim, Akuli, Arinnao and Akuli semi-linked up and have sent Ben a DM with the channel.

**Comment by Akuli** [comment]
Yes, there is now a Porcupine discord. For now I will be inviting people individually rather than opening it to the public. I think we have more or less everyone except rdbende there.

> I still don't get the hype around Discord

I would prefer IRC, but it has the huge downside of forgetting messages. Quite annoying even with a bot that remembers them (used to be a thing on ##learnpython).

> a decent (not Electron based) desktop client

A few hints:
- I think custom clients aren't really a thing because of how their API works. The API is only good for making bots AFAIK.
- Use `discord.com/app` in browser, "desktop client" is really just a browser window that goes there. I don't like it either.

My setup is:
- `discord.com/app` bookmarked in browser
- a browser extension that runs a custom javascript when entering discord, for some crude customization.

**Comment by rdbende** [comment]
> Also, you have stated before finding trouble in finding important notifications due to my mass-commenting problem and the many GH actions.

Yeah, that's a thing :)

**Comment by Moosems** [comment]
I just have the Discord phone app so I can run code and GH on my computer and message on my phone. This allows me to save battery and memory on my computer which is hard to charge and use my phone as a battery mule because I can easily charge it.

**Comment by Moosems** [comment]
> Yeah, that's a thing :)

I feel bad as it is, ok XD? I'm working on it.

**Comment by rdbende** [comment]
![image](https://github.com/Akuli/porcupine/assets/77941087/4cf2c840-93a4-43ce-bb0c-d86f943da076)

**Comment by Moosems** [comment]
Yay!

**Comment by Moosems** [comment]
> I feel bad as it is, ok XD? I'm working on it.

Well todays commits completely turned this into a lie.

**Comment by rdbende** [comment]
> and if I can find a decent (not Electron based) desktop client

In the end I found this:
![image](https://github.com/Akuli/porcupine/assets/77941087/9e8c0dfa-c82e-4981-aca0-76d83535946e)

**Comment by Moosems** [comment]
> > and if I can find a decent (not Electron based) desktop client
> 
> 
> 
> In the end I found this:
> 
> ![image](https://github.com/Akuli/porcupine/assets/77941087/9e8c0dfa-c82e-4981-aca0-76d83535946e)
> 
> 

What is that?
